src/kaboom.ts

```diff
@@ -3952,6 +3954,44 @@ export default (gopt: KaboomOpt = {}): KaboomCtx => {
 
 	}
 
+	function polygon(pts: Vec2[], opt: PolygonCompOpt = {}): PolygonComp {
+		if(pts.length < 3) throw new Error(`Polygon's need more than two points, ${pts.length} points provided`)
+		return {
+			id: "polygon",
+			pts,
+			draw(this: GameObj<PolygonComp>) {
+				drawPolygon(Object.assign(getRenderProps(this), Object.assign(opt,{
+					pts,
+				})))
+			},
+			renderArea(this: GameObj<AnchorComp | PolygonComp>) {
```

Comment:
Polygon is supported for collision detection, you can use 
```js
return new Polygon(this.pts)
```
here

Reply:
I believe I've done the thing! I also updated the `polygon` example to show the collision matching the polygon shape!